projects/step-frontend/src/lib/modules/grid/components/token-list/token-list.component.html

```diff
@@ -0,0 +1,103 @@
+<section class="header">
+  <step-btn-group>
+    <step-autorefresh-toggle [interval]="5000" (refresh)="loadTable()"> </step-autorefresh-toggle>
+  </step-btn-group>
+</section>
+<step-table [dataSource]="searchableToken$" [inProgress]="inProgress" matSort [matSortDisableClear]="true">
+  <ng-container matColumnDef="id" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Id</th>
+    <td mat-cell *matCellDef="let element">
+      {{ element.token!.id! }}
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="type" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Type</th>
+    <td mat-cell *matCellDef="let element">
+      <span class="token-type" [class]="'token-type-'.concat(element.token!.attributes!.$agenttype)"></span>
+      {{ TYPE_LABEL_TRANSLATIONS[element.token!.attributes!.$agenttype] }}
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="agent" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Agent</th>
+    <td mat-cell *matCellDef="let element">
+      {{ element.agent!.agentUrl! }}
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="attributes" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef>Attributes</th>
+    <td mat-cell *matCellDef="let element">
+      {{ _flatObjectFormatService.format(element.token!.attributes!) }}
```

Comment:
Code invocation in binding.
Introduce additional pipe, that invokes FlatObjectFormatService.
Or you can turn FlatObjectFormatService into pipe itslef in the following way
```
@Pipe({
  name:  'flatObjectFormat'
})
class FtabObjectFormatPipe implements PipeTransform {
  
  // static method to use it outside of templates
  static transform(obj: any): string {
       return JSON.stringify(obj)
                 .replace(/{/g, '')
                 .replace(/}/g, '')
                 .replace(/:/g, '=')
                 .replace(/,/g, ', ')
                 .replace(/"/g, '');
  }

  transform(obj: any): string {
     return FtabObjectFormatPipe.transform(obj);
  }

}
```

Reply:
thanks for the suggestion with the static method, since I need it in the controller as well

Reply:
I not going to use methods in the template any more, but more (pure) pipes